src/ParametersHandler/YarpImplementation/src/YarpImplementation.cpp

```diff
@@ -58,18 +150,20 @@ YarpImplementation::weak_ptr YarpImplementation::getGroup(const std::string& nam
         return m_lists.at(name);
     }
 
-    return YarpImplementation::make_shared();
+    return std::make_shared<YarpImplementation>();
 }
 
-void YarpImplementation::setGroup(const std::string &name, IParametersHandler<YarpImplementation>::shared_ptr newGroup)
+void YarpImplementation::setGroup(const std::string& name, IParametersHandler::shared_ptr newGroup)
 {
-    auto downcastedPtr = std::dynamic_pointer_cast<YarpImplementation>(newGroup); //to access m_container
+    auto downcastedPtr = std::dynamic_pointer_cast<YarpImplementation>(newGroup); // to access
```

Comment:
This is an issue now. Before we were sure that downcasting was working. Now we cannot be sure anymore. Maybe the check should not be a simple ``assert``, but rather an if clause. If the downcasting is not valid, we return immediately, maybe with an error message.

I wonder if now it makes sense to have a bool return value instead :thinking:

Reply:
If it is not possible to cast the pointer a runtime error will be thrown.

Reply:
probably a `static_pointer_cast` could be used. I have to check

Reply:
It does not throw a runtime error, it returns an empty pointer and the assert works only on debug.

Static cast does not work because of inheritance.

Reply:
b9e043d41456629b7be2a5b9fabab29af781a11e substitute the `dynamic_pointer_cast` with `static_pointer_cast`. The tests runs happily

Reply:
This does not fix the problem. You can still pass the pointer to a different implementation. Before you would have had a null pointer. Now you would have a malformed pointer. In both cases segfault. 

Check https://stackoverflow.com/questions/1254777/static-cast-versus-dynamic-cast

Reply:
Ok so a boolean should solve the problem

Reply:
This is what I wrote in the first comment 😜

Reply:
Done in https://github.com/dic-iit/bipedal-locomotion-controllers/pull/28/commits/21d5ed943bfbe7b364896ede6487ea634800cd11